providers/enchant_nuspell.cpp

```diff
@@ -0,0 +1,445 @@
+/* enchant
+ * Copyright (C) 2020 Sander van Geloven
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.	 See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ *
+ * In addition, as a special exception, Dom Lachowicz
+ * gives permission to link the code of this program with
+ * non-LGPL Spelling Provider libraries (eg: a MSFT Office
+ * spell checker backend) and distribute linked combinations including
+ * the two.  You must obey the GNU General Public License in all
+ * respects for all of the code used other than said providers.  If you modify
+ * this file, you may extend this exception to your version of the
+ * file, but you are not obligated to do so.  If you do not wish to
+ * do so, delete this exception statement from your version.
+ */
+
+/*
+ * This is the Nuspell Enchant Backend.
+ * Nuspell is by Dimitrij Mijoski and Sander van Geloven.
+ * See: http://nuspell.github.io/
+ */
+
+#include "config.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <string>
+#include <vector>
+
+#include "enchant-provider.h"
+#include "unused-parameter.h"
+
+#include <boost/locale/utf8_codecvt.hpp>
```

Comment:
Is it really necessary to use Boost? I don't really want to add any dependencies that the spell-checker backend does not need, and I see that the nuspell library does not require Boost.

Reply:
Fixed in https://github.com/nuspell/enchant/commit/205c4b8423d3140a1c6f3e3e77f0841145c7fb69#diff-b4a4b7f548e6522299f21f4f5632ff1cL48 and https://github.com/nuspell/enchant/commit/205c4b8423d3140a1c6f3e3e77f0841145c7fb69#diff-b4a4b7f548e6522299f21f4f5632ff1cL268

Reply:
This was a remnant of the initial implementation with Nuspell 2.3, with Nuspell 3.0 this is the way to do it indeed.